packages/dropdowns/src/styled/items/StyledItem.ts

```diff
@@ -26,13 +27,15 @@ export const getItemPaddingVertical = (props: IStyledItemProps & ThemeProps<Defa
 };
 
 const getColorStyles = (props: IStyledItemProps & ThemeProps<DefaultTheme>) => {
+  const color = props.isDanger
+    ? getColor('dangerHue', 600, props.theme)
+    : props.theme.colors.foreground;
+
   return css`
     background-color: ${props.isFocused &&
     !props.disabled &&
-    getColor('primaryHue', 600, props.theme, 0.08)};
-    color: ${props.disabled
-      ? getColor('neutralHue', 400, props.theme)
-      : props.theme.colors.foreground};
+    getColor(props.isDanger ? 'dangerHue' : 'primaryHue', 600, props.theme, 0.08)};
+    color: ${props.disabled ? getColor('neutralHue', 400, props.theme) : color};
```

Comment:
Let's extract more of this into local variables for readability. Something like (untested):

```js
let foregroundColor;
let backgroundColor;

if (props.disabled) {
  foregroundColor = getColor('neutralHue', 400, props.theme);
} else if (props.isDanger) {
  foregroundColor = getColor('dangerHue', 600, props.theme);
  backgroundColor = props.isFocused && rgba(foregroundColor, 0.08);
} else {
  foregroundColor = props.theme.colors.foreground;
  backgroundColor = props.isFocused && getColor('primaryHue', 600, props.theme, 0.08); 
}

return css`
  background-color: ${backgroundColor};
  color: ${foregroundColor};
`;
```

The rgba function comes from `import { rgba } from 'polished'`;